e2e/etcd_test.go

```diff
@@ -172,14 +210,16 @@ type etcdProcess struct {
 type etcdProcessConfig struct {
 	args        []string
 	dataDirPath string
-	acurl       url.URL
+	acurl       []string
 	isProxy     bool
 }
 
 type etcdProcessClusterConfig struct {
-	clusterSize   int
-	proxySize     int
-	isClientTLS   bool
+	clusterSize int
+	proxySize   int
+	isClientTLS bool
+	// serve both TLS and non-TLS
+	isClientBoth  bool
```

Comment:
`isClientTLS` and `isClientBoth` should probably be merged into one enum field `clientTLS` with values `None, TLS, Both`